### ISSUE TITLE
No error for missing method call 

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#35557]
**TypeScript Version:**  3.8.0-dev.20191206

**Search Terms:**
- diagnostics
- conditional
- missing function call

**Code**

```ts
import * as fs from 'fs';

fs.stat('/path/to/file', function(err, stats) {
    if (stats.isDirectory) { 
        console.log(`[Directory] ${stats.ctime}`)
    }
});
```

**Expected behavior:**
Error on `isDirectory`. This is a method but it has not been called

**Actual behavior:**
No error reported.

If I remove the console log line

**Comment by MartinJohns** [comment]
This will only report an error if `strictNullChecks` is enabled, and `stats.isDirectory` is a non-nullable function. See #33178 and #9041.

**Comment by mjbvz** [comment]
This is in strict mode:

![Dec-07-2019 09-36-01](https://user-images.githubusercontent.com/12821956/70378423-06d5d880-18d5-11ea-9d04-785b8b59cd81.gif)

**Comment by RyanCavanaugh** [comment]
It looks like we're confused by any reference to `stats` in the body if the `if`

**Comment by dragomirtitian** [comment]
The way the check is done for usage seems brittle. The only check that is done is for symbol `id` equality: 

```ts
const functionIsUsedInBody = forEachChild(ifStatement.thenStatement, function check(childNode): boolean | undefined {
    if (isIdentifier(childNode)) {
        const childSymbol = getSymbolAtLocation(childNode);
        if (childSymbol && childSymbol.id === testedFunctionSymbol.id) { // HERE 
            return true;
        }
    }

    return forEachChild(childNode, check);
});
```
This has the drawback of not actually working at all for `symbols` which have not had an id assigned (and we get `undefined === undefined`).  This could be fixed by using `getSymbolId` which would assign the id, or by direct object equality (since as far as I understand it the same symbol object should be used and `id` equality when assignment at this location seems no better) but this fix would not fix the following surprising behavior: 

```ts
interface StatsBase {
    isDirectory(): boolean;
}
function A(stats2: StatsBase, stats: StatsBase) {
    if (stats.isDirectory) { // no error because we access isDirectory on stat2
        stats2.isDirectory(); // comment this and we get an error above
    }
}
```

[Playground Link](https://www.typescriptlang.org/play/#code/JYOwLgpgTgZghgYwgAgMpjmAzgITllAbwFgAoZC5YLAEWCggTAHsoBPACgEoAuZAI2bMANhDggA3GQC+ZGAFcQTYMxDIAghywZsAJj7pMufBAA0ybUYM7jBLshLlKwGMi02AdNToMmrNvaEyAD0wcggzMjQUKwCjHDyBMgA7iiISFhYVLT0jCzsyKoWOrpklOXFRrpeOb75nFwSIWEIzAC2bRDgyGAAFtTI4gAmKSgA5hBgg2rRsXCCAG4QZZSypNJAA)

The symbols `isDirectory` is used, in the `if` condition and in the body but the target is different.

Proposed solution: 
1. Use `getSymbolId` for when comparing symbol ids
2. Recursively compare the property access chain so that all the symbols in the chain are the same.
